### PR TITLE
Select two redesign

### PR DESCRIPTION
**Pull request by MadonnaMat**
Fixes Issue #1

### CODE REVIEW

File: SelectTwo.elm [hunk 1]
```diff
@@ -215,6 +191,8 @@ setSearch filter =
         map (\s -> { s | search = search })
 
 
+{-| a
```
Comment (crazymykl) [inline SelectTwo.elm]: Document this better.

File: SelectTwo.elm [hunk 2]
```diff
@@ -265,4 +243,88 @@ selectGroup msg list =
...
+-}
+defaultsFromList : List msg -> List (GroupSelectTwoOption msg) -> List (SelectTwoOption msg)
+defaultsFromList defaults list =
+    list
+        |> List.concatMap (\( _, l ) -> l)
+        |> List.filter
+            (\l ->
+                l
+                    |> Tuple3.first
+                    |> Maybe.map (\a -> List.member a defaults)
```
Comment (crazymykl) [inline SelectTwo.elm]: The lambda could be `(flip List.member defaults)`, if you'd like.

File: SelectTwo.elm (hunk 1 above)
Comment (MadonnaMat) [reply SelectTwo.elm]: what? Your crazy? it's perfect! What?

**Comment by roovo** [comment]
just having a look at using this - nice to see it's being updated - like the look of what you're doing :)

Just one thing whilst you're in there is could you remember to update dependencies whilst you're at it -> you're using an older version of List Extra and it'd be handy (for me of course) if it was updated.

Library looks pretty cool so thanks for developing

**Comment by roovo** [comment]
you're a ⭐️  - thank you

**Pull request merged**